logcli 2.7.3 failed to run with go1.20

👋 while [regression testing go1.20 upgrade](https://github.com/Homebrew/homebrew-core/pull/122082), I ran into the following error:

```
level=info ts=2023-02-02T03:58:16.722819Z caller=loki.go:461 msg="Loki started"
==> /opt/homebrew/Cellar/logcli/2.7.3/bin/logcli --addr=http://localhost:62567 labels
panic: Something in this program imports go4.org/unsafe/assume-no-moving-gc to declare that it assumes a non-moving garbage collector, but your version of go4.org/unsafe/assume-no-moving-gc hasn't been updated to assert that it's safe against the go1.20 runtime. If you want to risk it, run with environment variable ASSUME_NO_MOVING_GC_UNSAFE_RISK_IT_WITH=go1.20 set. Notably, if go1.20 adds a moving garbage collector, this program is unsafe to use.

goroutine 1 [running]:
go4.org/unsafe/assume-no-moving-gc.init.0()
	go4.org/unsafe/assume-no-moving-gc@v0.0.0-20220617031537-928513b29760/untested.go:25 +0x178
```

full build log, https://pipelines.actions.githubusercontent.com/serviceHosts/d065b819-d194-4baa-bc84-7b4a294d8693/_apis/pipelines/1/runs/564571/signedlogcontent/19?urlExpires=2023-02-03T17%3A06%3A38.0355252Z&urlSigningMethod=HMACV1&urlSignature=pSX%2FY%2FAHvb9NFrLSm1xQHLgwZ%2FwQ5NinBMs1z6lh5sI%3D

same for `Promtail` as well

```
2023-02-02T05:40:26.2856970Z ##[error]==> Testing promtail
panic: Something in this program imports go4.org/unsafe/assume-no-moving-gc to declare that it assumes a non-moving garbage collector, but your version of go4.org/unsafe/assume-no-moving-gc hasn't been updated to assert that it's safe against the go1.20 runtime. If you want to risk it, run with environment variable ASSUME_NO_MOVING_GC_UNSAFE_RISK_IT_WITH=go1.20 set. Notably, if go1.20 adds a moving garbage collector, this program is unsafe to use.

goroutine 1 [running]:
go4.org/unsafe/assume-no-moving-gc.init.0()
	go4.org/unsafe/assume-no-moving-gc@v0.0.0-20220617031537-928513b29760/untested.go:25 +0x178
==> curl -s localhost:49370/metrics

==> Testing promtail (again)
panic: Something in this program imports go4.org/unsafe/assume-no-moving-gc to declare that it assumes a non-moving garbage collector, but your version of go4.org/unsafe/assume-no-moving-gc hasn't been updated to assert that it's safe against the go1.20 runtime. If you want to risk it, run with environment variable ASSUME_NO_MOVING_GC_UNSAFE_RISK_IT_WITH=go1.20 set. Notably, if go1.20 adds a moving garbage collector, this program is unsafe to use.
```

--- a/operator/go.mod
+++ b/operator/go.mod
@@ -146,7 +146,7 @@ require (
 	go.uber.org/multierr v1.7.0 // indirect
 	go.uber.org/zap v1.21.0 // indirect
 	go4.org/intern v0.0.0-20211027215823-ae77deb06f29 // indirect
-	go4.org/unsafe/assume-no-moving-gc v0.0.0-20220617031537-928513b29760 // indirect
+	go4.org/unsafe/assume-no-moving-gc v0.0.0-20230209150437-ee73d164e760 // indirect
 	golang.org/x/crypto v0.0.0-20220315160706-3147a52a75dd // indirect
 	golang.org/x/mod v0.6.0-dev.0.20220419223038-86c51ed26bb4 // indirect
 	golang.org/x/net v0.0.0-20220722155237-a158d28d115b // indirect

